bvm/ballerina-runtime/src/test/java/io/ballerina/runtime/test/config/ConfigTest.java

```diff
@@ -258,4 +258,25 @@ public void testEnvVarUnsupportedErrors(String variableName, Type type, String e
         }
 
     }
+
+    @Test
+    public void testCliWhenUnsupportedTypesWithinToml() {
+        ArrayType arrayType = TypeCreator.createArrayType(TYPE_STRING);
+        VariableKey v1 = new VariableKey(module, "v1", PredefinedTypes.TYPE_INT, true);
+        Type v2Type = new BIntersectionType(module, new Type[]{arrayType, PredefinedTypes.TYPE_READONLY},
+                arrayType, 0, true);
+        VariableKey v2 = new VariableKey(module, "v2", v2Type, true);
+        RuntimeDiagnosticLog diagnosticLog = new RuntimeDiagnosticLog();
+        ConfigResolver configResolver =
+                new ConfigResolver(Map.ofEntries(Map.entry(module, new VariableKey[]{v1, v2})), diagnosticLog,
+                        List.of(new CliProvider(ROOT_MODULE, "-CmyOrg.test_module.v1=87"),
+                                new TomlFileProvider(ROOT_MODULE,
+                                        getConfigPath("UnsupportedCLITypeConfig.toml"), Set.of(module))));
+        Map<VariableKey, ConfigValue> configValueMap = configResolver.resolveConfigs();
+        Object v1Value = configValueMap.get(v1).getValue();
+        Object v2Value = configValueMap.get(v2).getValue();
```

Comment:
Shall we use `configValueMap.get(v1).getValue()` inside the assertion calls? Creating new variables can be redundant if we don't use them multiple times.

Reply:
I agree. I'll fix that.